src/Aspire.Hosting/Dcp/Process/ProcessUtil.cs

```diff
@@ -34,6 +34,11 @@ public static (Task<ProcessResult>, IAsyncDisposable) Run(ProcessSpec processSpe
             EnableRaisingEvents = true
         };
 
+        if (!processSpec.InheritEnv)
+        {
+            process.StartInfo.Environment.Clear();
+        }
```

Comment:
Does this work? Is the environment populated when this is called?

Reply:
Yeah, it starts out with `process.StartInfo.Environment` containing the full environment of the parent process.